Subject: Add support for libpng 1.5.x and later

This commit fixes the build problem which is caused by new libpng

---
diff --git a/png.c b/png.c
index dadf817..3d7e797 100644
--- a/png.c
+++ b/png.c
@@ -69,7 +69,7 @@ int fh_png_load(char *name,unsigned char *buffer, unsigned char ** alpha,int x,i
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
@@ -161,7 +161,7 @@ int fh_png_getsize(char *name,int *x,int *y)
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
